crates/wasmtime/src/runtime/vm/traphandlers.rs

```diff
@@ -823,6 +825,43 @@ impl CallThreadState {
     /// executors as `Handler::resume` will be used.
     unsafe fn unwind(&self, store: &mut dyn VMStore) {
         let unwind = self.unwind.replace(UnwindState::None);
+
+        #[cfg(feature = "debug")]
+        match &unwind {
+            UnwindState::UnwindToHost {
+                reason: UnwindReason::Trap(TrapReason::Exception),
+                ..
+            } => {
+                use crate::store::AsStoreOpaque;
+                let exn = store
+                    .as_store_opaque()
+                    .pending_exception_owned_rooted()
+                    .expect("exception should be set when we are throwing");
+                store.block_on_debug_handler(crate::DebugEvent::ThrownException(exn.clone()));
+                store.block_on_debug_handler(crate::DebugEvent::UncaughtException(exn));
```

Comment:
Two events? Wouldn't `UncaughtException` always imply the former `ThrownException`?

Reply:
My thought process that led here was: I initially wanted events on caught and uncaught exceptions; but we don't yet have the mechanisms to give a view of state at the catch-point, only the throw-point, so the former became "thrown exception"; but then the exception is thrown whether it's caught in Wasm or uncaught.

I suppose I could rename `ThrownException` to `ThrownExceptionToWasm` or something similar, but then an observer who wants to see all exception throws needs to watch two different events.

I do agree it's a little awkward to see two distinct hooks at one static point; maybe we just document what `ThrownException` means and invoke (only) that here...

Reply:
OK, renamed to `CaughtExceptionThrown` and `UncaughtExceptionThrown` in 7e9d38686a.